### PR TITLE
Sarvesh/mosa 14 ost price oracle compiler warning

### PR DESCRIPTION
**Pull request by sarvesh-ost**
Fixed compiler warning 
1. changing constructor syntax 
2. Adding emit statement when emitting events 
3. Removed redundant OpsManaged statement 
4. Upgraded truffle version to 4.1.18 to support latest solc compiler syntax

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ before_script:
   - sudo redis-server /etc/redis/redis.conf --port 6380 --requirepass 'my-secret'
   - bash contracts/compile.sh
 script:
+  - truffle version
```
Comment (abhayks1) [inline .travis.yml]: @sarvesh-ost it's not needed

File: package.json [hunk 2]
```diff
@@ -30,7 +30,7 @@
     "mustache": "2.3.0"
   },
   "devDependencies": {
-    "truffle": "4.0.1",
+    "truffle": "v4.1.8",
```
Comment (abhayks1) [inline package.json]: why v4.1.8?

File: .travis.yml (hunk 1 above)
Comment (sarvesh-ost) [reply .travis.yml]: Yes, I will remove this. It was added to check truffle and solc version on Travis.

**Comment by abhayks1** [comment]
@benjaminbollen @jasonklein do we need to upgrade truffle/solidity version in openst protocol also?

File: package.json (hunk 2 above)
Comment (sarvesh-ost) [reply package.json]: Truffle 4.0.1 internally uses solc version 0.4.18. Which uses the old syntax of solidity. Upgrading truffle version to v4.1.8 also upgrades newer syntax of solidity(0.4.23).

**Comment by benjaminbollen** [comment]
@abhayks1 looks like @sarvesh-ost addressed your concerns?  Merging

**Pull request merged**